include/boost/gil/image_processing/fast_feature_detector.hpp

```diff
@@ -0,0 +1,170 @@
+                return -1;
+            else if (intensity > hi)
+                return 1;
+            else
+                return 0;
+        });
+    std::transform(
+        intensity_array.begin(),
+        intensity_array.end(),
+        back_inserter(threshold_indicator),
+        [low = I_p - t, hi = I_p + t](auto const& intensity) {
+            if (intensity < low)
+                return -1;
+            else if (intensity > hi)
+                return 1;
+            else
+                return 0;
+        });
+    //high speed test for eliminating non-corners
+    for (int i = 0; i <= 6; i += 2)
+    {
+        if (threshold_indicator[i] == 0 && threshold_indicator[i + 8] == 0)
+            return false;
+    }
+    for (int i = 1; i <= 7; i += 2)
+    {
+        if (threshold_indicator[i] == 0 && threshold_indicator[i + 8] == 0)
+            return false;
+    }
+    //final segment test
+    bool is_feature_point =
+        threshold_indicator.end() !=
+            std::search_n(threshold_indicator.begin(), threshold_indicator.end(), 9, -1) ||
+        threshold_indicator.end() !=
+            std::search_n(threshold_indicator.begin(), threshold_indicator.end(), 9, 1);
+    return is_feature_point;
+}
+template <typename srcview>
+std::ptrdiff_t
+    calculate_score(srcview& src, int i, int j, std::vector<point_t>& points, int threshold)
+{
+    int score = threshold;
+    std::ptrdiff_t low = threshold;
+    std::ptrdiff_t high = 255;
+    //score measure used= highest threshold for which a corner remains a corner. The cornerness of a corner decreases with increasing threshold
+    while (high - low > 1)
+    {
+        int mid = (low + high) / 2;
+        if (fast_feature_detector(src, i, j, points, mid))
+        {
+            low = mid;
+            score = std::max(score, mid);
+        }
+        else
+        {
+            high = mid - 1;
+        }
+    }
+    return low - 1;
+}
+}  // namespace detail
+template <typename srcview>
+void fast(
+    srcview& src,
+    std::vector<point_t>& keypoints,
+    std::vector<int>& scores,
+    bool nonmax = true,
+    int threshold = 10)
+{
+    //coordinates of a bresenham circle of radius 3
+    std::vector<point_t> final_points_clockwise{
+        point_t(3, 0),
+        point_t(3, 1),
+        point_t(2, 2),
+        point_t(1, 3),
+        point_t(0, 3),
+        point_t(-1, 3),
+        point_t(-2, 2),
+        point_t(-3, 1),
+        point_t(-3, 0),
+        point_t(-3, -1),
+        point_t(-2, -2),
+        point_t(-1, -3),
+        point_t(0, -3),
+        point_t(1, -3),
+        point_t(2, -2),
+        point_t(3, -1)};
+    //FAST features only calculated on grayscale images
+    auto input_image_view = color_converted_view<gray8_pixel_t>(src);
+    gray8_image_t FAST_image(src.dimensions());
+    // scores to be used during nonmaximum suppression
+    gray8_view_t FAST_SCORE_MATRIX = view(FAST_image);
+    fill_pixels(FAST_SCORE_MATRIX, gray8_pixel_t(0));
```

Comment:
Please, avoid such UPPERCASE names, unless it's a macro.
Avoid macros too.

Reply:
Done